docs/init.rst

```diff
@@ -301,6 +301,20 @@ Arguably, you can abuse converters as one-argument validators:
    ValueError: invalid literal for int() with base 10: 'x'
 
 
+If a converter's first argument has a type annotation, that type will appear in the signature for ``__init__``.
+A converter will override an explicit type annotation or ``type`` argument.
```

Comment:
nit: This changes current behaviour even if converters do not have type annotations. Though, these changes apply to `typing.get_type_hints(cls.__init__)`, not `typing.get_type_hints(cls)` and I'm not sure anybody will actually get anything broken by that.

Reply:
Hm if that's so important, maybe it would warrant a section in `types.rst`. Changelogs are terrible for long-term knowledge keeping. But I don't want to put that on poor @nosewings.